### PR TITLE
build: libgit2 makefile fixes

### PR DESCRIPTION
**Pull request by vtjnash**
Ensure that operations are happening at the right time, in the right places, in the right order, and with the correct dependencies.

### CODE REVIEW

File: deps/pcre.mk [hunk 1]
```diff
@@ -23,7 +23,7 @@ $(BUILDDIR)/pcre2-$(PCRE_VER)/build-compiled: $(BUILDDIR)/pcre2-$(PCRE_VER)/buil
 	$(MAKE) -C $(dir $<) $(LIBTOOL_CCLD)
 	echo 1 > $@
 
-$(BUILDDIR)/pcre2-$(PCRE_VER)/checked: $(BUILDDIR)/pcre2-$(PCRE_VER)/build-compiled
+$(BUILDDIR)/pcre2-$(PCRE_VER)/build-checked: $(BUILDDIR)/pcre2-$(PCRE_VER)/build-compiled
```
Comment (staticfloat) [inline deps/pcre.mk]: Is this change intentional?  It seems unrelated.

Comment (vtjnash) [reply deps/pcre.mk]: It is. It was as missed update in 406af98ea16, but I guess nobody _really_ missed it.
